libs/methodologies/bold/rule-processors/mass-id/avoided-emissions/src/avoided-emissions.processor.ts

```diff
@@ -81,21 +91,42 @@ export class AvoidedEmissionsProcessor extends RuleDataProcessor {
   }
 
   protected evaluateResult(ruleSubject: RuleSubject): EvaluateResultOutput {
-    const { emissionFactor, massIdDocumentValue } = ruleSubject;
+    const {
+      exceedingEmissionCoefficient: exceedingemissionCoefficient,
+      massIdDocumentValue,
+      wasteGeneratorBaseline,
+      wasteSubtype,
+    } = ruleSubject;
+
+    if (!isNonZeroPositive(exceedingemissionCoefficient)) {
+      return {
+        resultComment: RESULT_COMMENTS.MISSING_EXCEEDING_EMISSION_COEFFICIENT,
+        resultStatus: RuleOutputStatus.REJECTED,
+      };
+    }
 
-    if (!isNonZeroPositive(emissionFactor)) {
+    if (isNil(wasteGeneratorBaseline)) {
       return {
-        resultComment: RESULT_COMMENTS.MISSING_EMISSION_FACTOR,
+        resultComment:
+          RESULT_COMMENTS.MISSING_RECYCLING_BASELINE_FOR_WASTE_SUBTYPE(
+            wasteSubtype,
+          ),
         resultStatus: RuleOutputStatus.REJECTED,
       };
     }
 
-    const avoidedEmissions = emissionFactor * massIdDocumentValue;
+    const avoidedEmissionsBaseline =
+      AVOIDED_EMISSIONS_BASELINE_MATRIX[wasteSubtype][wasteGeneratorBaseline];
+
+    const avoidedEmissions =
+      (1 - exceedingemissionCoefficient) *
+      avoidedEmissionsBaseline *
```

Comment:
I know this is a big name, but I would like to be more clear about what this variable means, such as:
```suggestion
      avoidedEmissionsByMaterialAndBaselinePerTon *
```

Instead of `material`, we could also use `wasteSubtype`.